File: Java/src/main/java/Solution801_900/Solution883.java
package Solution801_900;

/**
 * @author windows
 */
public class Solution883 {
    public int projectionArea(int[][] grid) {
        int sum = 0;
        for (int i = 0; i < grid.length; i++) {
            int rowMax = 0;
            int colMax = 0;
            for (int j = 0; j < grid[i].length; j++) {
                rowMax = Math.max(rowMax, grid[i][j]);
                colMax = Math.max(colMax, grid[j][i]);
                if (grid[i][j] != 0){
                    sum++;
                }
            }
            sum += rowMax + colMax;
        }
        return sum;
    }


    public static void main(String[] args) {
        Solution883 solution883 = new Solution883();
    }
}

File: C++/src/Solution801_900/Solution883.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution883.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution::projectionArea(vector<vector<int>> &grid) {
    int m = grid.size();
    int n = grid[0].size();
    int left = 0;
    int before = 0;
    int counter = 0;
    for (int i = 0; i < m; ++i) {
        int max_val_left = grid[i][0];
        for (int j = 1; j < n; ++j) {
            max_val_left = max(max_val_left, grid[i][j]);
            if (grid[i][j]) counter++;
        }
        left += max_val_left;
    }
    for (int i = 0; i < n; ++i) {
        int max_val_before = grid[0][i];
        for (int j = 0; j < m; ++j) {
            max_val_before = max(max_val_before, grid[j][i]);
        }
        before += max_val_before;
    }
    return counter + left + before;
}
